### DISCUSSION TITLE
SmartAutoClicker-2.0.0

### DISCUSSION THREAD

**Discussion by Nain57** [Nain57/Smart-AutoClicker#228] [Announcements]
This is the first major overhaul for the application since its beginning. 

# New UI:
* The application is now following most of the Material Design 3 guidelines
* Screen are adapted for a portrait and landscape orientations
* Light/Dark mode are supported
* Android 13 dynamic colors (application colors will adapt to your device wallpaper colors)
* Tips and explanations are displayed below most of the fields
* Missing required field are now highlighted

# New features:
* Event can now be enabled/disabled when configuring your scenario
* New action: event state can be changed during scenario execution
* Action randomization: randomize a bit your actions values in order to lure anti detection software
* Increase condition capture minimum size
* Various bugfixes

<hr /><em>This discussion was created from the release <a href='https://github.com/Nain57/Smart-AutoClicker/releases/tag/2.0.0'>SmartAutoClicker-2.0.0</a>.</em>

**Comment by 12ChuAnh05** [comment]
I found 2 bugs
1. Cannot edit scenario after stopping
2. Cannot stop after starting the second time (Stop and start again)

**Reply by shmaungms** [reply]
I don't have either of the above bugs occurring to me. My phone made me uninstall the previous version before installing 2.0.0. I created a backup and imported it in to 2.0.0 with no problems. It will take me a little while to get used to the new UI.

I have one small bug in 10 minutes of use so far. Some text is partly cut off in portrait mode. 
![Screenshot_2023-02-23-13-28-04-46_b783bf344239542886fee7b48fa4b892](https://user-images.githubusercontent.com/120090978/220893918-4f328b69-ea66-4668-9f86-bbdce5c18dcf.jpg)

Also is this your app? https://play.google.com/store/apps/details?id=com.gameclicker.autoclicker.pro 
It's 99% the same as your previous version and until recently had the same landscape bug you fixed a while ago.
I wouldn't like someone stealing your code as this app is great!

**Reply by Nain57** [reply]
> I found 2 bugs
> 
> 1. Cannot edit scenario after stopping
> 2. Cannot stop after starting the second time (Stop and start again)

I can't reproduce your issues.
Can you be more specific ? (Phone brand, Android version, maybe screenshots of the issue)

**Reply by 12ChuAnh05** [reply]
> > I found 2 bugs
> > 
> > 1. Cannot edit scenario after stopping
> > 2. Cannot stop after starting the second time (Stop and start again)
> 
> I can't reproduce your issues. Can you be more specific ? (Phone brand, Android version, maybe screenshots of the issue)

My device is Galaxy S9+, rom ArrowOS 12.1. I think i will reinstall the program to try to fix the problem

**Reply by Nain57** [reply]
> I don't have either of the above bugs occurring to me. My phone made me uninstall the previous version before installing 2.0.0. I created a backup and imported it in to 2.0.0 with no problems. It will take me a little while to get used to the new UI.
> 
> I have one small bug in 10 minutes of use so far. Some text is partly cut off in portrait mode. ![Screenshot_2023-02-23-13-28-04-46_b783bf344239542886fee7b48fa4b892](https://user-images.githubusercontent.com/120090978/220893918-4f328b69-ea66-4668-9f86-bbdce5c18dcf.jpg)
> 
> Also is this your app? https://play.google.com/store/apps/details?id=com.gameclicker.autoclicker.pro It's 99% the same as your previous version and until recently had the same landscape bug you fixed a while ago. I wouldn't like someone stealing your code as this app is great!

Hi, thanks for your feedback !

No this app is not mine, someone has stolen my code and I would not recommend using it. Using this version can be really dangerous, as with the permissions requested by the app, everything you do on your device can be seen, recorded and sent to a server.

I have planned to publish the 2.0.0 version on the PlayStore myself, stay tuned !

**Comment by TeH07** [comment]
![Screenshot_20230224-134722.png](https://user-images.githubusercontent.com/86117427/221175538-211fb860-2dd0-4fa8-9f46-f49a8a1b8586.png)

I found this bug it not full screen and can not scroll. show only half button.

My phone is Sony xperia 1 screen display size 21:9

**Reply by shmaungms** [reply]
I had the same happen but I loaded my backup scenarios and forgot about it. 
And just looking at my phone the create new scenario is partly cut off too. 
![Screenshot_2023-02-24-16-33-10-11_58a85b8371fe7005d9929f266090d4d6](https://user-images.githubusercontent.com/120090978/221205048-c0d6435d-0a48-4eee-a905-b7ba521a9225.jpg)

and playing a little more the very bottom scenario is therefore also partly hidden by the menu and back keys too when I scroll to the bottom of the page. A newly created scenario is barely visible at all.

and more bugs: When scrolling up and down the new scenario button disappears usually when scrolling to the bottom, but not always. More often than not when scrolling back to the top it doesn't reappear when I scroll back to the top again, but if I go up and down enough times it reappears.

**Reply by Nain57** [reply]
Thanks for the feedback.
This will be fixed in the next version.
